website/docs/reference/resource-properties/constraints.md

```diff
@@ -518,3 +518,68 @@ alter table schema_name.my_model add constraint 472394792387497234 check (id > 0
 </div>
 
 </WHCode>
+
+### Custom constraints on models for advanced configuration of tables
+
+In dbt Cloud, you can use custom constraints on models for the advanced configuration of tables. Different data warehouses will support different syntax and capabilities, but custom constraints will always allow you to add configuration to specific columns.
+
+For example, custom constraints allow you to set [masking policies](https://docs.snowflake.com/en/user-guide/security-column-intro#what-are-masking-policies) in Snowflake when using a Create Table As Select (CTAS).
+
+[Databricks](https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-ddl-create-table-using.html) and [BigQuery](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#column_name_and_column_schema) are also other examples of data warehouses having their own set of parameters that can be set for columns in their CTAS statements.
+
+There are different scenarios of implementing contracts and constraints:
+
+- [Tag-based masking policies](https://github.com/dbt-labs/dbt-adapters/issues/85#issuecomment-2310772737).
+- [Add a masking policy without a tag](https://github.com/dbt-labs/dbt-adapters/issues/85#issuecomment-2314547162).
+
+Here's an example of how to implement tag-based masking policies with contracts and constraints using the following syntax:
+
+<File name='models/constraints_example.yml'>
+
+```yaml
+
+models:
```

Comment:
hmm not sure if this is the right version but the yaml is still formatted incorrectly. i would recommend checking out this page: https://docs-getdbt-rfi9dmjcm-dbt-labs.vercel.app/reference/resource-properties/constraints#defining-constraints to see how the yaml is independent and follow that indentation maybe? sorry if the yaml formatter websites didn't flag this but i do think the yaml isn't formatted right.

Reply:
Hiya @mirnawong1 

I've checked the YAML is ok for both examples on a few sites and they appear to be correct. Would you have a recommended source / page I can use to tripe check by chance?

Kind Regards
Natalie